tests/components/pilight/test_init.py

```diff
@@ -66,351 +61,340 @@ def set_callback(self, function):
+    with assert_setup_component(4):
+        whitelist = {
+            "protocol": [PilightDaemonSim.test_message["protocol"]],
+            "id": [PilightDaemonSim.test_message["message"]["id"]],
+        }
+        assert await async_setup_component(
+            hass, pilight.DOMAIN, {pilight.DOMAIN: {"whitelist": whitelist}}
+        )
+
+        await hass.async_start()
+        await hass.async_block_till_done()
+
+        expected_message = dict(
+            {
+                "protocol": PilightDaemonSim.test_message["protocol"],
+                "uuid": PilightDaemonSim.test_message["uuid"],
+            },
+            **PilightDaemonSim.test_message["message"],
+        )
+        debug_log_call = mock_debug.call_args_list[-3]
+
+        # Check if all message parts are put on event bus
+        for key, value in expected_message.items():
+            assert str(key) in str(debug_log_call)
+            assert str(value) in str(debug_log_call)
+
+
+@patch("pilight.pilight.Client", PilightDaemonSim)
+@patch("homeassistant.core._LOGGER.debug")
+async def test_whitelist_or_match(mock_debug, hass):
+    """Check whitelist filter with several subsection, should work."""
+    with assert_setup_component(4):
+        whitelist = {
+            "protocol": [
+                PilightDaemonSim.test_message["protocol"],
+                "other_protocol",
+            ],
+            "id": [PilightDaemonSim.test_message["message"]["id"]],
+        }
+        assert await async_setup_component(
+            hass, pilight.DOMAIN, {pilight.DOMAIN: {"whitelist": whitelist}}
+        )
+
+        await hass.async_start()
+        await hass.async_block_till_done()
+
+        expected_message = dict(
+            {
+                "protocol": PilightDaemonSim.test_message["protocol"],
+                "uuid": PilightDaemonSim.test_message["uuid"],
+            },
+            **PilightDaemonSim.test_message["message"],
+        )
+        debug_log_call = mock_debug.call_args_list[-3]
+
+        # Check if all message parts are put on event bus
+        for key, value in expected_message.items():
+            assert str(key) in str(debug_log_call)
+            assert str(value) in str(debug_log_call)
+
+
+@patch("pilight.pilight.Client", PilightDaemonSim)
+@patch("homeassistant.core._LOGGER.debug")
+async def test_whitelist_no_match(mock_debug, hass):
+    """Check whitelist filter with unmatched data, should not work."""
+    with assert_setup_component(4):
+        whitelist = {
+            "protocol": ["wrong_protocol"],
+            "id": [PilightDaemonSim.test_message["message"]["id"]],
+        }
+        assert await async_setup_component(
+            hass, pilight.DOMAIN, {pilight.DOMAIN: {"whitelist": whitelist}}
+        )
+
+        await hass.async_start()
+        await hass.async_block_till_done()
+        debug_log_call = mock_debug.call_args_list[-3]
+
+        assert not ("Event pilight_received" in debug_log_call)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)